src/common/file_system.cpp

```diff
@@ -841,4 +841,130 @@ bool FileSystem::IsRemoteFile(const string &path, string &extension) {
 	return false;
 }
 
+//===----------------------------------------------------------------------===//
+// FileHandleWrapper implementation
+//===----------------------------------------------------------------------===//
+FileHandleWrapper::FileHandleWrapper(FileSystem &file_system, unique_ptr<FileHandle> handle, bool flag)
+    : FileHandle(file_system, handle->GetPath(), handle->GetFlags()), internal_handle(std::move(handle)),
+      flag(flag) {
+	// Copy logger from internal handle if it exists
+	if (internal_handle->logger) {
+		logger = internal_handle->logger;
+	}
+}
+
+FileHandleWrapper::~FileHandleWrapper() {
```

Comment:
This is the key part: without the wrapper indirection, file handle instances cannot call virtual function `Close` at destructor.